src/main/scala/firrtl/ir/IR.scala

```diff
@@ -22,20 +22,45 @@ case object NoInfo extends Info {
   override def toString: String = ""
   def ++(that: Info): Info = that
 }
-case class FileInfo(info: StringLit) extends Info {
-  override def toString: String = " @[" + info.serialize + "]"
+
+/** Stores the string of a file info annotation in its escaped form. */
+case class FileInfo(escaped: String) extends Info {
+  override def toString: String = " @[" + escaped + "]"
   //scalastyle:off method.name
   def ++(that: Info): Info = if (that == NoInfo) this else MultiInfo(Seq(this, that))
+  def unescaped: String = FileInfo.unescape(escaped)
+}
+
+object FileInfo {
+  @deprecated("Use FileInfo.fromUnEscaped instead. FileInfo.apply will be removed in FIRRTL 1.5.", "FIRRTL 1.4")
+  def apply(info: StringLit): FileInfo = new FileInfo(escape(info.string))
+  def fromEscaped(s: String): FileInfo = new FileInfo(s)
+  def fromUnescaped(s: String): FileInfo = new FileInfo(escape(s))
+  /** prepends a `\` to: `\`, `\n`, `\t` and `]` */
+  def escape(s: String): String = {
+    s.replace("\\", "\\\\")
+      .replace("\n", "\\\n")
+      .replace("\t", "\\\t")
+      .replace("]", "\\]")
+  }
+  /** removes the `\` in front of `\`, `\n`, `\t` and `]` */
+  def unescape(s: String): String = {
+    s.replace("\\]", "]")
+      .replace("\\\t", "\t")
+      .replace("\\\n", "\n")
+      .replace("\\\\", "\\")
+  }
```

Comment:
I think this escaping logic is sufficient (and arguably better) for in and out of FIRRTL (although we should check that with the Parser), but I'm a bit worried about the Verilog emission. If you have a unicode character in a source locator, currently we emit a unicode escape, eg.
```
y <= x @[file.scala\t123:11 😎]
```
Emits as
```
assign y = x; // @[file.scala\t123:11 \uD83D\uDE0E]
```

This PR changes it to emit as:
```
assign y = x; // @[file.scala\t123:11 😎]
```

This may be legal, but if we're going to change behavior there should be tests illustrating this and we need to be sure that this doesn't violate the Verilog 1995 spec.

Reply:
I honestly completely forgot about the Verilog emission and need to fix that.
Thanks a lot for pointing that out!

The way I would do it is to call the `unescape` function and then the old `encodeJava` function in the Verilog emitter. I will add a commit shortly.